tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+
+
+@pytest.mark.include_capsule
+def test_positive_procedure_by_tag_check_migrations(sat_maintain):
+    """Run pre-migrations and post-migrations using advanced procedure by-tag
+
+    :id: 65cacca0-f142-4a63-a644-01f76120f16c
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain advanced procedure by-tag pre-migrations
+        2. Run satellite-maintain advanced procedure by-tag post-migrations
+
+    :expectedresults: procedures of pre-migrations tag and post-migrations tag should work.
+    """
+    result = sat_maintain.cli.AdvancedByTag.pre_migrations()
+    assert 'FAIL' not in result.stdout
+    rules = sat_maintain.execute('nft list tables')
+    assert 'FOREMAN_MAINTAIN_TABLE' in rules.stdout
+
+    result = sat_maintain.cli.AdvancedByTag.post_migrations()
+    assert 'FAIL' not in result.stdout
+    rules = sat_maintain.execute('nft list tables')
+    assert 'FOREMAN_MAINTAIN_TABLE' not in rules.stdout
+
+
+@pytest.mark.include_capsule
+def test_positive_procedure_by_tag_restore_confirmation(sat_maintain):
+    """Run restore_confirmation using advanced procedure by-tag
+
+    :id: f9e10352-04fb-49ba-8346-5b02e64fd028
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain advanced procedure by-tag restore
+
+    :expectedresults: procedure restore_confirmation should work.
+    """
+    result = sat_maintain.cli.AdvancedByTag.restore(options={'assumeyes': True})
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+def test_positive_sync_plan_with_hammer_defaults(request, sat_maintain):
+    """Verify that sync plan is disabled and enabled with hammer defaults set.
+
+    :id: b25734c8-470f-4cad-bc56-5c0f75aa7499
+
+    :steps:
+        1. Setup hammer on system with defaults set
+        2. Run satellite-maintain advanced procedure run sync-plans-disable
+        3. Run satellite-maintain advanced procedure run sync-plans-enable
+
+    :expectedresults: sync plans should get disabled and enabled.
+    """
+    sat_maintain.cli.Defaults.add({'param-name': 'organization_id', 'param-value': 1})
+
+    result = sat_maintain.cli.Advanced.run_sync_plans_disable()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+    result = sat_maintain.cli.Advanced.run_sync_plans_enable()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+    @request.addfinalizer
+    def _finalize():
+        sat_maintain.cli.Defaults.delete({'param-name': 'organization_id'})
+
+
+def test_positive_satellite_repositories_setup(sat_maintain):
+    """Verify that all required repositories gets enabled.
+
+    :id: e32fee2d-2a1f-40ed-9f94-515f75511c5a
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run repositories-setup --version 6.y
+
+    :BZ: 1684730, 1869731
+
+    :expectedresults: Required Satellite repositories for install/upgrade should get enabled
+    """
+    supported_versions = ['6.11', '6.12']
```

Comment:
You should also check if the current XY satellite version is any of them and skip the Test else this test would keep on running in future sat versions.

Reply:
Yes, will take this into different PR as mentioned earlier